internal/server/handler_v2.go

```diff
@@ -16,22 +16,64 @@
 package server
 
 import (
+	"bytes"
 	"context"
+	"io"
+	"net/http"
 
+	"github.com/datacommonsorg/mixer/internal/merger"
+	"github.com/datacommonsorg/mixer/internal/server/resource"
 	v1e "github.com/datacommonsorg/mixer/internal/server/v1/event"
 	v2 "github.com/datacommonsorg/mixer/internal/server/v2"
 	v2e "github.com/datacommonsorg/mixer/internal/server/v2/event"
-	v2observation "github.com/datacommonsorg/mixer/internal/server/v2/observation"
 	v2p "github.com/datacommonsorg/mixer/internal/server/v2/properties"
 	v2pv "github.com/datacommonsorg/mixer/internal/server/v2/propertyvalues"
 	"github.com/datacommonsorg/mixer/internal/server/v2/resolve"
 	"github.com/datacommonsorg/mixer/internal/util"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/reflect/protoreflect"
 
 	pbv2 "github.com/datacommonsorg/mixer/internal/proto/v2"
 )
 
+func fetchRemote(
+	metadata *resource.Metadata,
+	apiPath string,
+	in protoreflect.ProtoMessage,
+	out protoreflect.ProtoMessage,
+) error {
+	url := metadata.RemoteMixerUrl + apiPath
+	jsonValue, _ := protojson.Marshal(in)
+	request, _ := http.NewRequest("POST", url, bytes.NewBuffer(jsonValue))
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("X-API-Key", metadata.RemoteMixerApiKey)
+
+	client := &http.Client{}
```

Comment:
It feels like 'client' should be a field in 'server', and it should be passed in to 'fetchRemote' as an argument, otherwise, we need to init http.Client every time calling fetchRemote(), which seems an anti-pattern.

Reply:
Done